openhexa/sdk/pipelines/parameter.py

```diff
@@ -125,7 +132,179 @@ def normalize(value: typing.Any) -> typing.Any:
         return value
 
 
-TYPES_BY_PYTHON_TYPE = {str: String, bool: Boolean, int: Integer, float: Float}
+class PostgreSQLConnectionType(ParameterType):
+    @property
+    def spec_type(self) -> str:
+        return "postgresql"
+
+    @property
+    def expected_type(self) -> typing.Type:
+        return PostgreSQLConnectionType
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
+        if not allow_empty and value == "":
+            raise ParameterValueError("Empty values are not accepted.")
+
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        return value
+
+
+class S3ConnectionType(ParameterType):
+    @property
+    def spec_type(self) -> str:
+        return "s3"
+
+    @property
+    def expected_type(self) -> typing.Type:
+        return S3ConnectionType
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
```

Comment:
This feels duplicated between connection parameter types. We might need an intermediary class or helper methods to avoid copy pasting that kind of code.